### PR TITLE
feat: Create gnap key pair once per session

### LINKED ISSUE #1720: Refactor : Create a key pair per session not on every refresh
1. Make key export to false 
2. Create key pair only if no key pair is there for a session 
3. Check how generate key and sign in subtle crypto are related and do they persist the key pair in the browser storage

### PR DESCRIPTION
**Pull request by talwinder50**
closes #1720

Signed-off-by: talwinder50 <talwinderkaur50@gmail.com>

### CODE REVIEW

File: cmd/wallet-web/src/mixins/gnap/gnap.js [hunk 1]
```diff
@@ -36,7 +24,7 @@ export async function gnapRequestAccess(signer) {
       start: ['redirect'],
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/mixins/gnap/gnap.js:24]: need followup : `store.getters()` is vuex specific, these vue specific code shouldn't be part of mixin package which is pure javascript.

File: cmd/wallet-web/src/mixins/gnap/store.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  callOnStore(function (store) {
+    store.add(gnapKeyPair);
+  });
+  return gnapKeyPair;
+}
+
+function fetchStoredGnapKeyPair() {
+  return new Promise(function (resolve) {
+    callOnStore(function (store) {
+      store.get(1).onsuccess = function (event) {
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/mixins/gnap/store.js:32]: what is `store.get(1)`?

File: cmd/wallet-web/src/mixins/gnap/store.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  // This works on all devices/browsers, and uses IndexedDBShim as a final fallback
+  const indexedDB =
+    window.indexedDB ||
+    window.mozIndexedDB ||
+    window.webkitIndexedDB ||
+    window.msIndexedDB ||
+    window.shimIndexedDB;
+
+  const open = indexedDB.open(dbName, 3);
+  console.log('coming in call on store', open);
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/mixins/gnap/store.js:59]: left over?

File: cmd/wallet-web/src/mixins/gnap/store.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    fn_(store);
+    // Close the db when the transaction is done
+    tx.oncomplete = function () {
+      db.close();
+    };
+  };
+}
+
+// TODO Issue-364 Add gnap create key pair logic to agent sdk library
+function createKeyPair() {
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/mixins/gnap/store.js:80]: follow up: do not hardcode keytype and other parameters.

File: cmd/wallet-web/src/mixins/gnap/store.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  return new Promise(function (resolve) {
+    callOnStore(function (store) {
+      store.get(1).onsuccess = function (event) {
+        return resolve(event.target.result);
+      };
+    });
+  });
+}
+
+export function clearGnapStoreData() {
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/mixins/gnap/store.js:39]: follow up: deleting whole database vs clearing data.
Remember: creating database everytime impacts performance.

File: cmd/wallet-web/src/router/index.js [hunk 6]
```diff
@@ -17,9 +17,7 @@ const router = createRouter({
 router.beforeEach(async (to, from, next) => {
   store.dispatch('agent/flushStore');
   if (to.path === '/gnap') {
-    // TODO Issue-1720 create keypair only once if not found
-    const gnapKeyPair = await createKeyPair();
-    store.dispatch('updateGnapKeyPair', gnapKeyPair);
+    const gnapKeyPair = await getGnapKeyPair();
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/router/index.js:20]: vuex store call shouldn't be inside mixin javascript logic,

Comment (talwinder50) [reply cmd/wallet-web/src/router/index.js:20]: This one is in index.js not in mixin package

Comment (sudeshrshetty) [reply cmd/wallet-web/src/router/index.js:20]: I was asking to bring it back to `index.js` from `await createKeyPair()`

File: cmd/wallet-web/src/mixins/gnap/gnap.js (hunk 1 above)
Comment (talwinder50) [reply cmd/wallet-web/src/mixins/gnap/gnap.js:17]: removed from mixin package

File: cmd/wallet-web/src/mixins/gnap/store.js (hunk 5 above)
Comment (talwinder50) [reply cmd/wallet-web/src/mixins/gnap/store.js:37]: updated only deleted the the key value pair inside the store

File: cmd/wallet-web/src/mixins/gnap/store.js (hunk 4 above)
Comment (talwinder50) [reply cmd/wallet-web/src/mixins/gnap/store.js:75]: Part of #364 Agent-sdk

**Review by birtony — APPROVED** [review]

**Review by sudeshrshetty — APPROVED** [review]

**Pull request merged**